### ISSUE TITLE
[Common Lisp] Implement new Concept Exercise: packages

### ISSUE THREAD

**Issue by verdammelt** [exercism/v3#1347]
This issue describes how to implement the `packages` concept exercise for the Common Lisp track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

The goal of this exercise is to teach the basics of the Common Lisp package system via `defpackage`.

## Learning objectives

- Know how to use `defpackage` to do the following:
  - export a symbol from a package
  - import a symbol from another package
  - shadow a symbol from another package
- Know how to access exported symbols from a package
- Know how to access unexported symbols from a pakage.

## Out of scope

- All the individual functions which do the work of `defpackage` such as `import`, `export`, `shadow` etc.
- `intern`
- `gensym`

## Concepts

- packages

## Prerequisites

- variables
- symbols
- functions

## Resources to refer to

- [Hyperspec chapter on Packages][hyperspec-packages]
- [Programming in the Large: Packages and Symbols][practical-common-lisp-packages]

### Hints

The strongest hints will be that much of needed code has already been done in previous exercises as stub code.

### After

- discussion of functions such as `import`, `export`, `shadow

## Representer

The representer will need to recognize the differences between exported and non-exported symbol reference. It will also need to analyze and represent defined packages so that the `defpackage` form can be represented and analyzed.

## Analyzer

The analyzer will need to be able to analyze the `defpackage` form.

## Implementing

The stub code could be handled in two ways:

1. entirely blank
2. no `defpackge` form but all the needed symbols defined.

The latter has the advantage that it reduces the prerequisite load.

If a `package` concept reference file has not yet been created the PR for this exercise should do so.

### Suggested exercise tasks

- Tests will access some exported symbols of a package
- Tests will access some unexported symbols of a package
- Student will need to export a symbol which has the same name as an existing `CL-USER` symbol (_i.e._ `list`).
- Student will need to access a symbol from another package in their exercise code (perhaps they write two packages, or perhaps they need to access a symbol such as `UIOP:EMPTYP`)

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

[hyperspec-packages]: http://www.lispworks.com/documentation/HyperSpec/Body/11_.htm
[practical-common-lisp-packages]: http://www.gigamonkeys.com/book/programming-in-the-large-packages-and-symbols.html

**Comment by TheLostLambda** [comment]
Overall, this looks really good! It looks like a bit of formatting might have gotten lost in the after section though.

Additionally, `functions` seems to be a somewhat broad prerequisite, would `function-definition` be along the lines of what you had in mind? I think that's what it's been called in other exercises (unless I've forgotten to update some).

Overall looks great!

**Comment by verdammelt** [comment]
Fixed - thanks.
